saw-central/src/SAWCentral/Value.hs

```diff
@@ -747,7 +758,56 @@ getSharedContext :: TopLevel SharedContext
 getSharedContext = TopLevel_ (rwSharedContext <$> get)
 
 getJavaCodebase :: TopLevel JSS.Codebase
-getJavaCodebase = TopLevel_ (asks roJavaCodebase)
+getJavaCodebase =
+  do
+    status <- TopLevel_ (gets rwJavaCodebase)
```

Comment:
`TopLevel` is also a `MonadReader TopLevelRO` instance, so you ought to be able to get rid of the explicit newtype constructor here:

```suggestion
    status <- gets rwJavaCodebase
```